pkg/resourceinterpreter/default/native/aggregatestatus_test.go

```diff
@@ -1148,3 +1150,37 @@ func Test_aggregateHorizontalPodAutoscalerStatus(t *testing.T) {
 		})
 	}
 }
+
+func Test_getAllDefaultAggregateStatusInterpreter(t *testing.T) {
+	expected := map[schema.GroupVersionKind]aggregateStatusInterpreter{
+		{Group: "apps", Version: "v1", Kind: "Deployment"}:                     aggregateDeploymentStatus,
+		{Group: "apps", Version: "v1", Kind: "StatefulSet"}:                    aggregateStatefulSetStatus,
+		{Group: "batch", Version: "v1", Kind: "Job"}:                           aggregateJobStatus,
+		{Group: "", Version: "v1", Kind: "Pod"}:                                aggregatePodStatus,
+		{Group: "", Version: "v1", Kind: "Service"}:                            aggregateServiceStatus,
+		{Group: "networking.k8s.io", Version: "v1", Kind: "Ingress"}:           aggregateIngressStatus,
+		{Group: "batch", Version: "v1", Kind: "CronJob"}:                       aggregateCronJobStatus,
+		{Group: "apps", Version: "v1", Kind: "DaemonSet"}:                      aggregateDaemonSetStatus,
+		{Group: "", Version: "v1", Kind: "PersistentVolume"}:                   aggregatePersistentVolumeStatus,
+		{Group: "", Version: "v1", Kind: "PersistentVolumeClaim"}:              aggregatePersistentVolumeClaimStatus,
+		{Group: "policy", Version: "v1", Kind: "PodDisruptionBudget"}:          aggregatePodDisruptionBudgetStatus,
+		{Group: "autoscaling", Version: "v2", Kind: "HorizontalPodAutoscaler"}: aggregateHorizontalPodAutoscalerStatus,
+	}
```

Comment:
There should have been a similar test function that was merged before. I rethought it. Maybe such a test is not what we want. We don't need to care about whether the value is the same or not. We just need to test for existence.

How do you thinks?

Reply:
I’ve implemented the following code based on your suggestion:
```Go
func Test_getAllDefaultAggregateStatusInterpreter(t *testing.T) {
	expectedKinds := []schema.GroupVersionKind{
		{Group: "apps", Version: "v1", Kind: "Deployment"},
		{Group: "apps", Version: "v1", Kind: "StatefulSet"},
		{Group: "batch", Version: "v1", Kind: "Job"},
		{Group: "", Version: "v1", Kind: "Pod"},
		{Group: "", Version: "v1", Kind: "Service"},
		{Group: "networking.k8s.io", Version: "v1", Kind: "Ingress"},
		{Group: "batch", Version: "v1", Kind: "CronJob"},
		{Group: "apps", Version: "v1", Kind: "DaemonSet"},
		{Group: "", Version: "v1", Kind: "PersistentVolume"},
		{Group: "", Version: "v1", Kind: "PersistentVolumeClaim"},
		{Group: "policy", Version: "v1", Kind: "PodDisruptionBudget"},
		{Group: "autoscaling", Version: "v2", Kind: "HorizontalPodAutoscaler"},
	}

	got := getAllDefaultAggregateStatusInterpreter()

	if len(got) != len(expectedKinds) {
		t.Errorf("getAllDefaultAggregateStatusInterpreter() length = %d, want %d", len(got), len(expectedKinds))
	}

	for _, key := range expectedKinds {
		_, exists := got[key]
		if !exists {
			t.Errorf("getAllDefaultAggregateStatusInterpreter() missing key %v", key)
		}
	}
}
```
This logic should work well as long as we only want to test that all expected `GroupVersionKind` entries exist in the result, without worrying about the specific values. This approach will be more resilient even if the underlying mapping logic changes, as it focuses on the presence of the keys rather than their corresponding values.
Let me know your thoughts!